### DISCUSSION TITLE
How do I run code once per server startup?

### DISCUSSION THREAD

**Discussion by oluwatimilehin** [remix-run/remix#8694] [Q&A] [answered]
I would like to run custom code that connects to a service anytime the server starts up. Is there a Remix equivalent to [instrumentation](https://github.com/vercel/next.js/discussions/15341#discussioncomment-7091594)?

**Comment by gustavopch** [comment]
~~Put the code in `entry.server.tsx`.~~
Edit: not really, the behavior of this file depends on many factors.

**Reply by oluwatimilehin** [reply]
I've created an `app/entry.server.ts` file and added a console.log line, which doesn't show up in the logs.

My understanding from [this](https://remix.run/docs/en/main/file-conventions/entry.server) and [this](https://stackoverflow.com/questions/76966992/how-to-run-initialization-code-only-once-when-the-remix-server-starts) is that the file is triggered for every HTTP request.

What I'd like is to see it in the startup logs like below, even before any request is made:
```

20:08:20 │ remix    │ [shopify-api/INFO] version 7.7.0, environment Remix
20:08:21 │ remix    │ Remix App Server started at http://localhost:49199 (http://192.168.1.239:49199)
20:08:21 │ remix    │  info  app server ready (1s)
20:08:21 │ remix    │  info  hmr
```

Is this possible with Remix?

**Reply by gustavopch** [reply]
Hmm, I think the behavior of `entry.server.tsx` will depend on many factors. What platform are you using to host your app? If you're using Express, `server.ts` should work.

**Reply by oluwatimilehin** [reply, marked as answer]
Ah thanks, I was using the [Shopify app template](https://www.npmjs.com/package/@shopify/shopify-app-remix#authenticating-admin-requests) and I had to put it in shopify.server.js.
